### PR TITLE
cellMsgDialogOpen: Go further using cellMsgDialogOpen2.

### PR DESCRIPTION
**Pull request by ikki84**
Notes:
Needs proper implementation (!)
The functions shows similarities but the bits in argument "type" may vary a little.
Comparing both functions but testing in NPJA00014 I believe it could be acceptable until it be implemented properly

Feel free to reject if it´s not allowed

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp [hunk 1]
```diff
@@ -124,6 +119,14 @@ s32 cellMsgDialogOpen2(u32 type, vm::cptr<char> msgString, vm::ptr<CellMsgDialog
 	return CELL_OK;
 }
 
+s32 cellMsgDialogOpen(u32 type, vm::cptr<char> msgString, vm::ptr<CellMsgDialogCallback> callback, vm::ptr<void> userData, vm::ptr<void> extParam)
+{
+	//Note: This function needs proper implementation, solve first argument "type" conflict with MsgDialogOpen2 in cellMsgDialog.h.
+	LOG_TODO(HLE, "cellMsgDialogOpen Unimplemented using cellMsgDialogOpen2 instead");
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: I'd like it use cellSysutil.todo() formatting style, print arguments and propagate return value.

File: rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp [hunk 2]
```diff
@@ -124,6 +119,13 @@ s32 cellMsgDialogOpen2(u32 type, vm::cptr<char> msgString, vm::ptr<CellMsgDialog
 	return CELL_OK;
 }
 
+s32 cellMsgDialogOpen(u32 type, vm::cptr<char> msgString, vm::ptr<CellMsgDialogCallback> callback, vm::ptr<void> userData, vm::ptr<void> extParam)
+{
+	//Note: This function needs proper implementation, solve first argument "type" conflict with MsgDialogOpen2 in cellMsgDialog.h.
+	cellSysutil.todo("cellMsgDialogOpen(type=0x%x, msgString=%s, callback=*0x%x, userData=*0x%x, extParam=*0x%x)", type, msgString, callback, userData, extParam);
+	return	cellMsgDialogOpen2(type, msgString, callback, userData, extParam);
```
Comment (danilaml) [inline rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: tab after return

Comment (Zangetsu38) [inline rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: @Nekotekina Problem with this change, now log indicate 3 time same line 

>·U {PPU[0x1000000] Thread (main_thread) [0x00611e70]} cellSysutil TODO: cellMsgDialogOpen(type=0x0, msgString=“This game supports the system data Auto Save function.
Do not power off the system when the HDD access indicator is flashing.”, callback=*0x642dc8, userData=*0x101d9720, extParam=*0x0)
·W {PPU[0x1000000] Thread (main_thread) [0x00611e70]} cellSysutil: cellMsgDialogOpen2(type=0x0, msgString=“This game supports the system data Auto Save function.
Do not power off the system when the HDD access indicator is flashing.”, callback=*0x642dc8, userData=*0x101d9720, extParam=*0x0)
·E {PPU[0x1000000] Thread (main_thread) [0x00611e70]} cellSysutil: This game supports the system data Auto Save function.
Do not power off the system when the HDD access indicator is flashing.

First have a 3 time line message for game 
>This game supports the system data Auto Save function.
Do not power off the system when the HDD access indicator is flashing.

And after have 2 time line message:
>type=0x0, msgString=“”, callback=*0x642dc8, userData=*0x101d9720, extParam=*0x0)

**Comment by ikki84** [comment]
@Nekotekina @danilaml Sorry for the late reply. I changed the file according to the recommendations.

**Comment by kd-11** [comment]
@ikki84 Rebase to get rid of the merge commit. It'll be requested before merging.

**Comment by Nekotekina** [comment]
@kd-11 well, it's easy to squash

File: rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp (hunk 2 above)
Comment (paulsapps) [inline rpcs3/Emu/Cell/Modules/cellMsgDialog.cpp]: Logged twice is expected since its just forwarding to another function

**Pull request merged**